### PR TITLE
add missing methods in js backend

### PR DESCRIPTION
**Pull request by grazil**
Hello,

I've found missing methods in JS backend.

Have a nice day.

### CODE REVIEW

File: lib/js/dom.nim [hunk 1]
```diff
@@ -507,6 +507,7 @@ proc replace*(loc: Location, s: cstring)
 proc back*(h: History)
 proc forward*(h: History)
 proc go*(h: History, pagesToJump: int)
+proc pushState*(h: History, stateObject, title, url: cstring)
```
Comment (dom96) [inline lib/js/dom.nim]: The ``stateObject`` isn't a really a string :)

Comment (grazil) [reply lib/js/dom.nim]: Yes, in theory, i agree...  ;-)

But according to this [page](https://developer.mozilla.org/en-US/docs/Web/API/History_API) :
> The state object can be anything that can be serialized.

And in practice, it work well with a string (cstring in Nim).

Comment (dom96) [reply lib/js/dom.nim]: I think a generic type would work better. Are there other functions in ``dom`` that take similar arguments?

Comment (grazil) [reply lib/js/dom.nim]: Ok, i think you're right for the generic type. I haven't found a function in the `dom` that take similar arguments. 

The `stateObject` is just a basic JavaScript object for passing user variables as in the example in the [mozilla page](https://developer.mozilla.org/en-US/docs/Web/API/History_API) : 

`var stateObj = { foo: "bar" };`
`history.pushState(stateObj, "page 2", "bar.html");`

So, it doesn't have predefined properties.

I have to admit i don't know how to properly implement this...

What about `proc pushState*[T](h: History, stateObject: T, title, url: cstring)` ? 

I hope i am not annoying you, i am a beginner in Nim but i try to evolve. :)

Comment (dom96) [reply lib/js/dom.nim]: Yeah, that should work (do test it out if you can :)). Don't worry, you're not annoying me :)

**Comment by dom96** [comment]
thanks

**Pull request merged**